### ISSUE TITLE
Should a history of losses and training losses be explicity returned 

### ISSUE THREAD

**Issue by ablaom** [JuliaAI/IterationControl.jl#47]
Currently the workflow is to define an external variable and use the "callback" control `WithLossDo` - but perhaps there's an argument for collecting a history of the loss (and training loss and iteration count?) and returning these.
